testing/go/schemas_test.go

```diff
@@ -0,0 +1,282 @@
+				Expected: []sql.Row{},
+			},
+			{
+				Query: "SELECT * FROM mySchema.test;",
+				Expected: []sql.Row{
+					{1, 1},
+					{2, 2},
+				},
+			},
+			{
+				Query: "CREATE TABLE otherSchema.test (pk BIGINT PRIMARY KEY, v1 BIGINT);",
+			},
+			{
+				Query:    "insert into otherSchema.test values (3,3), (4,4)",
+				Expected: []sql.Row{},
+			},
+			{
+				Query: "SELECT * FROM otherSchema.test;",
+				Expected: []sql.Row{
+					{3, 3},
+					{4, 4},
+				},
+			},
+		},
+	},
+	{
+		Name: "insert, update, delete with schema",
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "create schema mySchema",
+			},
+			{
+				Query: "create schema otherSchema",
+			},
+			{
+				Query: "CREATE TABLE mySchema.test (pk BIGINT PRIMARY KEY, v1 BIGINT);",
+			},
+			{
+				Query: "CREATE TABLE otherSchema.test (pk BIGINT PRIMARY KEY, v1 BIGINT);",
+			},
+			{
+				Query: "insert into mySchema.test values (1,1), (2,2)",
+			},
+			{
+				Query:    "insert into otherSchema.test values (3,3), (4,4)",
+				Expected: []sql.Row{},
+			},
+			{
+				Query: "update mySchema.test set v1 = 3 where pk = 1",
+			},
+			{
+				Query: "update otherSchema.test set v1 = 4 where pk = 3",
+			},
+			{
+				Query: "delete from mySchema.test where pk = 2",
+			},
+			{
+				Query: "delete from otherSchema.test where pk = 4",
+			},
+			{
+				Query: "SELECT * FROM mySchema.test;",
+				Expected: []sql.Row{
+					{1, 3},
+				},
+			},
+			{
+				Query: "SELECT * FROM otherSchema.test;",
+				Expected: []sql.Row{
+					{3, 4},
+				},
+			},
+		},
+	},
+	{
+		Name: "schema does not exist",
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "create schema mySchema",
+			},
+			{
+				Query: "CREATE TABLE mySchema.test (pk BIGINT PRIMARY KEY, v1 BIGINT);",
+			},
+			{
+				Query:       "CREATE TABLE otherSchema.test (pk BIGINT PRIMARY KEY, v1 BIGINT);",
+				ExpectedErr: true,
+			},
+			{
+				Query: "insert into mySchema.test values (1,1), (2,2)",
+			},
+			{
+				Query:       "insert into otherSchema.test values (3,3), (4,4)",
+				ExpectedErr: true,
+			},
+			{
+				Query: "SELECT * FROM mySchema.test;",
+				Expected: []sql.Row{
+					{1, 1},
+					{2, 2},
+				},
+			},
+			{
+				Query:       "SELECT * FROM otherSchema.test;",
+				ExpectedErr: true,
+			},
+		},
+	},
+	{
+		Name: "create new database and new schema",
+		Skip: true,
+		SetUpScript: []string{
+			"CREATE DATABASE db2;",
+			"USE db2;", // TODO: not a real postgres statement
+			"create schema schema2;",
+			"use postgres",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "CREATE TABLE db2.schema2.test (pk BIGINT PRIMARY KEY, v1 BIGINT);",
+			},
+			{
+				Query: "SELECT * FROM db2.schema2.test;",
+				Expected: []sql.Row{
+					{1, 1},
+					{2, 2},
+				},
+			},
+		},
+	},
+	// More tests:
+	// * table name resolution among different schemas
+	// * alter table statements, when they work better
+	// * revision qualifiers
+	// * drop schema
+	// * more statement types
```

Comment:
More Testing Ideas:
* `INSERT INTO schema1 SELECT FROM schema2`
* Subqueries accessing different schemas in the same `SELECT`
* Joins across schemas
* Table names matching schema names. For example, `test1.test1`, `test1.test2`, `test2.test1`, `test2.test2` (more for future robustness)
* We don't have AS OF yet, so it may require the alternate syntax (not sure if it works yet), but reference a schema that isn't in the working set (was removed), but is in the commit history

Reply:
For that last one, I guess that needs `DROP SCHEMA` support. Perhaps could add that as a skipped test. Can still test across the commit history though on non-dropped schemas.